av/filter/graph.pyx

```diff
@@ -143,10 +145,46 @@ cdef class Graph(object):
 
         return self.add('buffer', args, name=name)
 
+    def add_abuffer(self, template=None, sample_rate=None, format=None, layout=None, channels=None, name=None):
```

Comment:
I haven't had a chance to review this actual pice of code, but in any case it deserves plenty of unit tests, especially considering all the `if` branches which raise exceptions: these need to be tested with assertRaises

Reply:
... or at the minimum a couple of integration tests. At least asserting that we haven't broken it horribly in the future.

Reply:
Sure, I'll add some tests. It might take a while though. Most likely will port C audio filtering examples.

Reply:
Added a filter_audio.py - ported a C demo. Touches most of the initialization logic.